src/Tests/dotnet.Tests/dotnet-workload-clean/GivenDotnetWorkloadClean.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System.CommandLine;
+using System.IO;
+using Xunit.Abstractions;
+using Xunit;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Utilities;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.Cli.Workload.Install.Tests;
+using ManifestReaderTests;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.DotNet.Workloads.Workload;
+using FluentAssertions;
+using Microsoft.DotNet.Workloads.Workload.Install;
+using Microsoft.DotNet.Workloads.Workload.Repair;
+using Microsoft.DotNet.Workloads.Workload.Clean;
+using Microsoft.DotNet.Workloads.Workload.List;
+using Microsoft.DotNet.Cli.Workload.List.Tests;
+using System.Collections.Generic;
+using System.CommandLine.Parsing;
+
+namespace Microsoft.DotNet.Cli.Workload.Clean.Tests
+{
+    public class GivenDotnetWorkloadClean : SdkTest
+    {
+        private readonly BufferedReporter _reporter;
+        private readonly string _manifestPath;
+
+        private MockWorkloadManifestUpdater _manifestUpdater = new();
+        private readonly string _sdkFeatureVersion = "6.0.100";
+        private readonly string _installingWorkload = "xamarin-android";
+        private readonly string dotnet = nameof(dotnet);
+        private readonly string _profileDirectoryLeafName = "user-profile";
+
+        public GivenDotnetWorkloadClean(ITestOutputHelper log) : base(log)
+        {
+            _reporter = new BufferedReporter();
+            _manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "Sample.json");
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        public void GivenWorkloadCleanFileBasedItRemovesPacksAndPackRecords(bool userLocal, bool cleanAll)
+        {
+            // Setup ( I wish we could not copy this everywhere, but these are convenient as locals. )
```

Comment:
Should probably remove/clean up this comment and then maybe file an issue if you think there's work required here.

Reply:
I don't think there's anything we can easily do, this is only copied two times and its just a lot of local variables that are needed for the test. Technically we could put it in a class but that'd end up being more loc/complexity than it currently is. (I tried it and got rid of it because it was worse.) The comment is basically an admission of 'this sucks, but dont waste time trying to make it better'